saleor/tests/e2e/checkout/shipping/test_use_external_shiping_methods_in_checkout.py

```diff
@@ -0,0 +1,136 @@
+from ...webhooks.utils import create_webhook
+from ..utils import (
+    checkout_complete,
+    checkout_create,
+    checkout_delivery_method_update,
+    checkout_dummy_payment_create,
+)
+
+
+@pytest.mark.vcr
+@pytest.mark.e2e
+def test_use_external_shipping_methods_in_checkout_core_1652(
+    e2e_not_logged_api_client,
+    e2e_staff_api_client,
+    permission_manage_products,
+    permission_manage_channels,
+    permission_manage_shipping,
+    permission_manage_product_types_and_attributes,
+    permission_manage_apps,
+    settings,
+):
+    # Before
+    settings.PLUGINS = [
+        "saleor.plugins.webhook.plugin.WebhookPlugin",
+        "saleor.payment.gateways.dummy.plugin.DummyGatewayPlugin",
+    ]
+    permissions = [
+        permission_manage_products,
+        permission_manage_channels,
+        permission_manage_shipping,
+        permission_manage_product_types_and_attributes,
+        permission_manage_apps,
+    ]
+    assign_permissions(e2e_staff_api_client, permissions)
+
+    (
+        warehouse_id,
+        channel_id,
+        channel_slug,
+        _shipping_method_id,
+    ) = prepare_shop(e2e_staff_api_client)
+
+    app_input = {"name": "external_shipping", "permissions": ["MANAGE_SHIPPING"]}
+    app_data = add_app(e2e_staff_api_client, app_input)
+    app_id = app_data["app"]["id"]
+
+    target_url = "http://localhost:8080"
+    webhook_input = {
+        "app": app_id,
+        "syncEvents": ["SHIPPING_LIST_METHODS_FOR_CHECKOUT"],
+        "asyncEvents": [],
+        "isActive": True,
+        "name": "external shipping methods",
+        "targetUrl": target_url,
+        "query": "subscription {\n  event {\n    ... on ShippingListMethodsForCheckout {\n      __typename\n      checkout{\n        shippingAddress{\n          country{\n            code\n          }\n        }\n      }\n    }\n  }\n}\n",
+        "customHeaders": "{}",
+    }
+    create_webhook(e2e_staff_api_client, webhook_input)
+
+    (
+        _product_id,
+        product_variant_id,
+        product_variant_price,
+    ) = prepare_product(
+        e2e_staff_api_client,
+        warehouse_id,
+        channel_id,
+        variant_price=9.99,
+    )
+    product_variant_price = float(product_variant_price)
+
+    # Step 1 - Create checkout for product
+    lines = [
+        {
+            "variantId": product_variant_id,
+            "quantity": 4,
+        },
+    ]
+    checkout_data = checkout_create(
+        e2e_not_logged_api_client,
+        lines,
+        channel_slug,
+        email="testEmail@example.com",
+        set_default_billing_address=True,
+        set_default_shipping_address=True,
+    )
+    checkout_id = checkout_data["id"]
+    assert checkout_data["isShippingRequired"] is True
+    calculated_subtotal = product_variant_price * 4
+    assert checkout_data["totalPrice"]["gross"]["amount"] == calculated_subtotal
+    shipping_methods = checkout_data["shippingMethods"]
+    assert len(shipping_methods) == 3
```

Comment:
You have 3 external shipping methods, but you assert only 2 of them - is it intended?

Reply:
I have two external shipping methods that I want to check as the should have come from webhook and one created in Saleor that's why I assert only externals :) I will add an assertion for that one too